# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/canvas/OffscreenCanvas.h
# Commit: 19fa8d4af66f
# Full Hash: 19fa8d4af66fb0dc9ace1dcf9cd365513573f05b
# Author: Jamie Nicol <jnicol@mozilla.com>
# Date: 2024-07-23 09:34:47
# Regressor Bug: 1898238
# File Overlap Count: 1
# Description:
#   Bug 1898238 - Don't call TexTypeForWebgl when initializing offscreen canvas. r=aosmond,jgilbert
#   
#   This is only used if the OffscreenCanvas has a webgl context, in which
#   case it forwards the tex type to WebGLContext::PresentFrontBuffer(). A
#   later patch in this series will make TexTypeForWebgl() return a
# ==============================================================================

diff -r 216ace67b0e2 -r 19fa8d4af66f dom/canvas/OffscreenCanvas.h
--- a/dom/canvas/OffscreenCanvas.h	Mon Jul 22 16:00:16 2024 +0000
+++ b/dom/canvas/OffscreenCanvas.h	Mon Jul 22 16:02:33 2024 +0000
@@ -48,15 +48,14 @@
   OffscreenCanvasCloneData(OffscreenCanvasDisplayHelper* aDisplay,
                            uint32_t aWidth, uint32_t aHeight,
                            layers::LayersBackend aCompositorBackend,
-                           layers::TextureType aTextureType, bool aNeutered,
-                           bool aIsWriteOnly, nsIPrincipal* aExpandedReader);
+                           bool aNeutered, bool aIsWriteOnly,
+                           nsIPrincipal* aExpandedReader);
   ~OffscreenCanvasCloneData();
 
   RefPtr<OffscreenCanvasDisplayHelper> mDisplay;
   uint32_t mWidth;
   uint32_t mHeight;
   layers::LayersBackend mCompositorBackendType;
-  layers::TextureType mTextureType;
   bool mNeutered;
   bool mIsWriteOnly;
   RefPtr<nsIPrincipal> mExpandedReader;
@@ -76,7 +75,6 @@
 
   OffscreenCanvas(nsIGlobalObject* aGlobal, uint32_t aWidth, uint32_t aHeight,
                   layers::LayersBackend aCompositorBackend,
-                  layers::TextureType aTextureType,
                   already_AddRefed<OffscreenCanvasDisplayHelper> aDisplay);
 
   void Destroy();
@@ -195,7 +193,6 @@
 
   layers::LayersBackend mCompositorBackendType =
       layers::LayersBackend::LAYERS_NONE;
-  layers::TextureType mTextureType = layers::TextureType::Unknown;
 
   RefPtr<OffscreenCanvasDisplayHelper> mDisplay;
   RefPtr<CancelableRunnable> mPendingCommit;